src/front/glsl_new/ast.rs

```diff
@@ -49,7 +49,47 @@ pub enum Profile {
 pub struct Context {
     pub expressions: Arena<Expression>,
     pub local_variables: Arena<LocalVariable>,
-    pub lookup_local_variables: FastHashMap<String, Handle<LocalVariable>>,
+    pub scopes: Vec<FastHashMap<String, Handle<LocalVariable>>>,
```

Comment:
This data structure is rather allocation-heavy. Maybe leave a TODO to make it more memory friendly?

Reply:
done